packages/sol-compiler/src/utils/compiler.ts

```diff
@@ -163,15 +159,19 @@ function printCompilationErrorsAndWarnings(solcErrors: solc.SolcError[]): void {
  * Gets the source tree hash for a file and its dependencies.
  * @param fileName Name of contract file.
  */
-export function getSourceTreeHash(resolver: Resolver, importPath: string): Buffer {
-    const contractSource = resolver.resolve(importPath);
-    const dependencies = parseDependencies(contractSource);
+export async function getSourceTreeHashAsync(
+    resolver: ResolverEngine<ImportFile>,
+    importPath: string,
+): Promise<Buffer> {
+    const imFile: ImportFile = await resolver.require(importPath);
```

Comment:
This looks odd. Could we go with `importFile` maybe? 
With variable name same as the type we could also drop the type altogether.